content-warehouse/src/main/java/contentwarehouse/v1/FetchAcl.java

```diff
@@ -0,0 +1,95 @@
+ * limitations under the License.
+ */
+
+package contentwarehouse.v1;
+
+// [START contentwarehouse_fetch_acl]
+import com.google.cloud.contentwarehouse.v1.DocumentName;
+import com.google.cloud.contentwarehouse.v1.DocumentServiceClient;
+import com.google.cloud.contentwarehouse.v1.DocumentServiceSettings;
+import com.google.cloud.contentwarehouse.v1.FetchAclRequest;
+import com.google.cloud.contentwarehouse.v1.FetchAclResponse;
+import com.google.cloud.contentwarehouse.v1.RequestMetadata;
+import com.google.cloud.contentwarehouse.v1.UserInfo;
+import com.google.cloud.resourcemanager.v3.Project;
+import com.google.cloud.resourcemanager.v3.ProjectName;
+import com.google.cloud.resourcemanager.v3.ProjectsClient;
+import java.io.IOException;
+
+
+public class FetchAcl {
+  public static void fetchAcl() throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-project-id";
+    String location = "your-location"; // Format is "us" or "eu".
+    String userId = "your-user-id"; // Format is user:<user-id>
+    String documentId = "your-documentid";
+    fetchAcl(projectId, location, userId, documentId);
+  }
+
+  public static void fetchAcl(String projectId, String location, String userId, String documentId) 
+      throws IOException {
+    String projectNumber = getProjectNumber(projectId); 
+    String endpoint = String.format("%s-contentwarehouse.googleapis.com:443", location);
+
+    DocumentServiceSettings documentServiceSettings = 
+        DocumentServiceSettings.newBuilder().setEndpoint(endpoint).build();
+        
+    try (DocumentServiceClient documentServiceClient = 
+            DocumentServiceClient.create(documentServiceSettings)) {
+
+      UserInfo userInfo = 
+          UserInfo.newBuilder().setId(userId).build();
+
+      DocumentName documentName = 
+          DocumentName.ofProjectLocationDocumentName(projectNumber, location, documentId);
+
+      RequestMetadata requestMetadata = 
+          RequestMetadata.newBuilder().setUserInfo(userInfo).build();
+
+      FetchAclResponse fetchAclResponse = 
+          FetchAclResponse.newBuilder().build();
+
+      if (documentId != null || documentId == "") { 
+        /* The full resource name of the document, e.g.:
+        projects/{project_number}/locations/{location}/documents/{document_id} */
+        FetchAclRequest fetchAclRequest = 
+            FetchAclRequest.newBuilder()
+                .setRequestMetadata(requestMetadata)
+                .setResource(documentName.toString()).build(); 
+        fetchAclResponse = documentServiceClient.fetchAcl(fetchAclRequest);
+      } else {
+        FetchAclRequest fetchAclRequest = 
+            FetchAclRequest.newBuilder()
+                .setRequestMetadata(requestMetadata)
+                .setResource(projectNumber)
+                .setProjectOwner(true).build();
+        fetchAclResponse = documentServiceClient.fetchAcl(fetchAclRequest);
+      }
```

Comment:
Instead of sending FetchAclRequest in the if else condition, wdyt about using the condition to create a new resource string and then outside the block, create the request?

Something like, 

```
String resource = "";

if (condition) {
 resource = documentName.toString();
} else {
 resource = projectNumber
}

.setResource(resource);
```

Reply:
I'm not sure that this would work due to .setProjectOwner(true), it is only applicable when the ACL request is for a project. It would throw an error if it's set on the document scope.